cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -133,41 +161,13 @@ func (impl *OSBuildJobImpl) Run(job worker.Job) error {
 		}
 	}
 
-	var r []error
-	var targetResults []*target.TargetResult
-
-	for _, t := range args.Targets {
-		switch options := t.Options.(type) {
-		case *target.LocalTargetOptions:
-			if !osbuildOutput.Success {
-				continue
-			}
-			var f *os.File
-			imagePath := path.Join(outputDirectory, exportPath, options.Filename)
-			if options.StreamOptimized {
-				f, err = vmware.OpenAsStreamOptimizedVmdk(imagePath)
-				if err != nil {
-					r = append(r, err)
-					continue
-				}
-			} else {
-				f, err = os.Open(imagePath)
-				if err != nil {
-					r = append(r, err)
-					continue
-				}
-			}
-
-			err = job.UploadArtifact(options.Filename, f)
-			if err != nil {
-				r = append(r, err)
-				continue
-			}
+	if len(args.Targets) == 0 {
+		// There is no upload target, mark this job a success.
+		osbuildJobResult.Success = true
+		osbuildJobResult.UploadStatus = "success"
+	} else if len(args.Targets) == 1 {
```

Comment:
I don't have a problem with being explicit here and have the `else if` branch, but it should not be needed because of the `if len(args.Targets) > 1 {` earlier.

Reply:
I know, but it feels easier to read for me, because it makes the intention clear. But I'm also ok removing it ....

Reply:
That's fine by me... :wink: